src/shared/codelens/pythonCodeLensProvider.ts

```diff
@@ -59,75 +72,209 @@ const getLambdaHandlerCandidates = async ({ uri }: { uri: vscode.Uri }): Promise
+    }
+    const logger = getLogger()
+
+    const [handlerFilePrefix, handlerFunctionName] = params.handlerName.split('.')
+    const debugHandlerFileName = `${handlerFilePrefix}_debug`
+    const debugHandlerFunctionName = 'lambda_handler'
+    // TODO: Sanitize handlerFilePrefix, handlerFunctionName, debugHandlerFunctionName
+    try {
+        logger.info('pythonCodeLensProvider.makeLambdaDebugFile params:', JSON.stringify(params, undefined, 2))
+        const template = `
+import ptvsd
+from ${handlerFilePrefix} import ${handlerFunctionName} as _handler
+
+
+def ${debugHandlerFunctionName}(event, context):
+    print('waiting for debugger to attach...')
+    ptvsd.enable_attach(address=('0.0.0.0', ${params.debugPort}), redirect_output=True)
+    ptvsd.wait_for_attach()
+    print('debugger attached')
+    return _handler(event, context)
+
+`
+
+        const outFilePath = path.join(params.outputDir, `${debugHandlerFileName}.py`)
+        logger.info('pythonCodeLensProvider.makeLambdaDebugFile outFilePath:', outFilePath)
+        await writeFile(outFilePath, template)
+
+        return {
+            outFilePath,
+            debugHandlerName: `${debugHandlerFileName}.${debugHandlerFunctionName}`
+        }
+    } catch (err) {
+        logger.error('makeLambdaDebugFile failed:', err as Error)
+        throw err
+    }
+}
+
+const fixFilePathCapitalization = (filePath: string): string => {
+    if (process.platform === 'win32') {
+        const startsWithLowercase = new RegExp(/^[a-z].*/)
+        if (startsWithLowercase.test(filePath)) {
+            return  filePath.slice(0, 1).toUpperCase() + filePath.slice(1)
+        }
+    }
+
+    return filePath
+}
+
+const makeDebugConfig = ({debugPort, samProjectCodeRoot}: {
+    debugPort?: number,
+    samProjectCodeRoot: string,
+}): PythonDebugConfiguration => {
+    return {
+        type: PYTHON_LANGUAGE,
+        request: 'attach',
+        name: 'SamLocalDebug',
+        host: 'localhost',
+        port: debugPort!,
+        pathMappings: [
+            {
+                // tslint:disable-next-line:no-invalid-template-strings
+                localRoot: fixFilePathCapitalization(samProjectCodeRoot),
+                remoteRoot: '/var/task'
+            }
+        ],
+    }
+}
+
+export async function initialize({
     configuration,
-    toolkitOutputChannel,
+    outputChannel: toolkitOutputChannel,
     processInvoker = new DefaultSamCliProcessInvoker(),
     taskInvoker = new DefaultSamCliTaskInvoker()
-}: CodeLensProviderParams): void {
+}: CodeLensProviderParams): Promise<void> {
     const logger = getLogger()
 
-    const runtime = 'python3.7' // TODO: Remove hard coded value
+    const runtime = 'python3.6' // TODO: Remove hard coded value
 
-    const invokeLambda = async (args: LambdaLocalInvokeArguments) => {
+    const invokeLambda = async (args: LambdaLocalInvokeParams) => {
         const samProjectCodeRoot = await getSamProjectDirPathForFile(args.document.uri.fsPath)
```

Comment:
For consistency & feedback, please add the following here, to parallel [localLambdaRunner](https://github.com/aws/aws-toolkit-vscode/blob/develop/src/shared/codelens/localLambdaRunner.ts#L75) - this lets the user know that we're preparing their app for run/debug

```
        // Switch over to the output channel so the user has feedback that we're getting things ready
        toolkitOutputChannel.show(true)

        toolkitOutputChannel.appendLine(
            localize(
                'AWS.output.sam.local.start',
                'Preparing to run {0} locally...',
                args.handlerName,
            )
        )

```

Reply:
Done.
✔️